docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
 # Developer Guide
 
+<!-- TOC -->
+
+- [Developer Guide](#developer-guide)
+    - [Acknowledgements](#acknowledgements)
+    - [Setting up](#setting-up)
+        - [Setting up the project in your computer](#setting-up-the-project-in-your-computer)
+        - [Before writing code](#before-writing-code)
+    - [Design](#design)
+        - [Architecture](#architecture)
+        - [General Sequence](#general-sequence)
+        - [UI component](#ui-component)
+        - [Parser component](#parser-component)
+        - [Accounts Component](#accounts-component)
+        - [Forex component](#forex-component)
+    - [Implementation](#implementation)
+        - [Create/Delete account feature](#createdelete-account-feature)
+        - [Delete-account feature](#delete-account-feature)
+        - [Add/Withdraw money feature](#addwithdraw-money-feature)
+        - [View balance feature](#view-balance-feature)
+        - [Show-rate feature](#show-rate-feature)
+        - [Money exchange feature](#money-exchange-feature)
+    - [Appendix: Requirements](#appendix--requirements)
+        - [Product scope](#product-scope)
+        - [Target user profile](#target-user-profile)
+        - [Value proposition](#value-proposition)
+        - [User Stories](#user-stories)
+        - [Non-Functional Requirements](#non-functional-requirements)
+        - [Glossary](#glossary)
+    - [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+  <!-- TOC -->
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
```

Comment:
Could you consider adding more details to this section? (similarly for "Non-functional requirements", "Glossary" and "Appendix: Instructions for manual testing")